web/src/js/__tests__/ducks/utils/storeSpec.js

```diff
@@ -0,0 +1,77 @@
+jest.unmock('../../../ducks/utils/store')
+
+import reduceStore, * as storeActions from '../../../ducks/utils/store'
+
+describe('store reducer', () => {
+    it('should return initial state', () => {
+        expect(reduceStore(undefined, {})).toEqual({
+            byId: {},
+            list: [],
+            listIndex: {},
+            view: [],
+            viewIndex: {},
+        })
+    })
+
+    it('should handle add action', () => {
+        let state = reduceStore(undefined, {})
+        expect(state = reduceStore(state, storeActions.add({id: 1}))).toEqual({
+                byId: { [1]: {id: 1} },
+                listIndex: { [1]: 0 },
+                list: [ {id: 1} ],
+                view: [ {id: 1} ],
+                viewIndex: { 1: 0 },
+        })
+
+        expect(reduceStore(state, storeActions.add({id: 9}))).toEqual({
+            byId: { [1]: {id:1}, [9]: {id:9} },
+            listIndex: { [1]: 0, [9]: 1 },
+            list: [ {id: 1}, {id: 9} ],
+            view: [ {id: 1}, {id: 9} ],
+            viewIndex: { 1: 0, 9: 1 },
+        })
```

Comment:
I think these tests would be more expressive if we'd assign the entries to variables, e.g. like this:
```javascript
let a = {id: 1},
    b = {id: 9}

expect(reduceStore(state, storeActions.add(b))).toEqual({
    byId: { 1: a, 9: b },
    listIndex: { 1: 0, 9: 1 },
    list: [ a, b ],
    view: [ a, b ],
    viewIndex: { 1: 0, 9: 1 },
})
```